### PR TITLE
Documentation/contribution section

### LINKED ISSUE #202: Documentation: Contributing section
Will link away from the 'pages' site.

Should have info about:
* Http server
* Docker
* GitHub Actions
* Pages

### PR DESCRIPTION
**Pull request by quirogas**
Create a contributing section for Documentation.

- Resolves #202

### CODE REVIEW

**Comment by MarkArbogast** [comment]
The `!!! quote` in the Tagging Convention section of docker.md isn't displaying correctly

File: contributing/docker.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Server Development
+# Base Adapter Images
```
Comment (MarkArbogast) [inline contributing/docker.md:1]: We need to explain why someone should read this page. It's not clear to me when someone would need this information.

File: contributing/http_server.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
+# HTTP Server
```
Comment (MarkArbogast) [inline contributing/http_server.md:1]: similarly here, we need to explain why someone would need this page's information. Maybe there should be a "Contributing" homepage that ties everything together.

**Review by MarkArbogast — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -63,6 +63,8 @@ If you wish to contribute code, and you have not signed our Contributor Licence
 to do so when you open a Pull Request. For any questions about the CLA process, please refer to our
 [FAQ](https://cla.vmware.com/faq).
 
+For additional information about contributing, go to the [contribution documentation](contributrion/README.md)
```
[the remark this answers is not in the document]
Comment (MarkArbogast) [reply README.md:66]: Also, should it be `contributing`?

File: contributing/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Contributing
+
+Before you start contributing to the project, you should familiarize yourself with the [documentation](vmware.github.io/vmware-aria-operations-integration-sdk/),
+[issue tracker](https://github.com/vmware/vmware-aria-operations-integration-sdk/issues), [project structure](index.md),
+and basic functioning of every component.
```
Comment (MarkArbogast) [inline contributing/README.md:5]: Rather than `every component`, it would be better to provide a specific list of pages someone should review.

**Review by MarkArbogast — CHANGES_REQUESTED** [review]
I like the README! Just needs a few tweaks.

File: contributing/README.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# Contributing
+
+Contributing to the project does not necessarily require familiarity with every component. To find a quick breakdown
+of the smaller parts of this project, consult the [index](index.md), which breaks down the SDK Structure into small
+components and references to help contributors find the necessary references without having to go in-depth into the
+project's overall structure. For a more in-depth breakdown of the project structure, it would be beneficial for you
+to have a look at the [documentation](vmware.github.io/vmware-aria-operations-integration-sdk/),
+[issue tracker](https://github.com/vmware/vmware-aria-operations-integration-sdk/issues),
+[project structure](index.md), and basic functioning of every component.
```
Comment (kjrokos) [inline contributing/README.md:9]: This is better, but now seems a bit wordy and redundant in places. It also links to the same page for a detailed breakdown as the overview page. Since we don't really have any more detailed information than what's linked from the index, I think contacting us is the best option. Maybe something like this?

```suggestion
Contributing to the project does not necessarily require familiarity with every component. The [index page](index.md) explains the project structure and links to some high-level project documentation. If you require detailed information not covered here, please open a [discussion](https://github.com/vmware/vmware-aria-operations-integration-sdk/discussions/new?category=general)
```

File: contributing/README.md [hunk 6]
```diff
@@ -1,23 +1,28 @@
...
-Before you start contributing to the project, you should familiarize yourself with the [documentation](vmware.github.io/vmware-aria-operations-integration-sdk/),
-[issue tracker](https://github.com/vmware/vmware-aria-operations-integration-sdk/issues), [project structure](index.md),
-and basic functioning of every component.
+Contributing to the project does not necessarily require familiarity with every component. To find a quick breakdown
+of the smaller parts of this project, consult the [index](index.md), which breaks down the SDK Structure into small
+components and references to help contributors find the necessary references without having to go in-depth into the
+project's overall structure. For a more in-depth breakdown of the project structure, it would be beneficial for you
+to have a look at the [documentation](vmware.github.io/vmware-aria-operations-integration-sdk/),
+[issue tracker](https://github.com/vmware/vmware-aria-operations-integration-sdk/issues),
+[project structure](index.md), and basic functioning of every component.
```
Comment (MarkArbogast) [inline contributing/README.md:9]: we shouldn't suggest that someone needs to look at every component to make a contribution.

**Review by MarkArbogast — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -63,7 +63,7 @@ If you wish to contribute code, and you have not signed our Contributor Licence
 to do so when you open a Pull Request. For any questions about the CLA process, please refer to our
 [FAQ](https://cla.vmware.com/faq).
 
-For additional information about contributing, go to the [contribution documentation](contributrion/README.md)
+For additional information about contributing, go to the [contribution section](contribution/README.md)
```
[the remark this answers is not in the document]
Comment (kjrokos) [reply README.md:66]: Looks like it should be `contributing` instead of `contribution`

File: contributing/README.md (hunk 6 above)
Comment (kjrokos) [reply contributing/README.md:9]: Yup, see my suggestion here: https://github.com/vmware/vmware-aria-operations-integration-sdk/pull/221#discussion_r1256325624

**Review by MarkArbogast — APPROVED** [review]

**Review by kjrokos — APPROVED** [review]
